### DISCUSSION TITLE
SFTP Verification Web Tool

### DISCUSSION THREAD

**Discussion by rdoering** [drakkan/sftpgo#1587] [General]
Huhu,

I am looking for a Web-Tool to verify SFTP connections.

Do you know the situation, when a customer tells you, that your service is unavailable, while your monitoring looks fine? In merely all cases it is an infrastructure problem at the customer site.

I would like to give the customer a way to verify/ investigate SFTP connection issues.

 Something like this tool for TLS
![image](https://github.com/drakkan/sftpgo/assets/1110248/ba0f8638-1bc5-44a7-b475-e9921f88cb60)

**Comment by rdoering** [comment]
It seems, that such a tool is not that popular :)

**Reply by thielj** [reply]
Probably because nobody wants to put their private key into a website ;)

A simple `ssh -v` usually gives some useful diagnostic output. 

Create a test account, upload a single file test.txt and ask the user to run `scp test@host.example.com:test.txt .`. Add -v if that doesn't work.

Curl has SFTP support as well.

There are also JavaScript based terminals that can run a ssh session in the browser. Check WeTTY, for example.

**Reply by rdoering** [reply]
Thanks for your response. 

For sure you are right, that this would be necessary to verify the logon process from end to end.  But that is not necessary to verify that 

- your SFTP service is reachable
- some options for example host key and ciphers are correct or as expected

**Reply by thielj** [reply]
@rdoering This works with SFTP as well: https://sshcheck.com/

**Reply by rdoering** [reply]
@thielj No, why I wasn't notified about your post? We discussed how to build such a tool and a colleague found your link too.  Now I want to paste this link here also and see, that you found it one week before.

Thank you very much. This tool is superp.

**Comment by rdoering** [comment]
Thanks @thielj
